### PR TITLE
feat(SHP-545): Create new address

### PR DESCRIPTION
**Pull request by ezelopes-mohc**

### CODE REVIEW

File: addresses.go [hunk 1]
```diff
@@ -4,21 +4,20 @@ import "github.com/MOHC-LTD/shopify"
 
 // AddressDTO represents a Shopify Address in HTTP requests and responses
 type AddressDTO struct {
-	Address1     string  `json:"address1,omitempty"`
```
Comment (Ianmohc) [inline addresses.go:7]: I would leave this one alone where it's always been. Customers have a specific address endpoint so the DTO could be reused from that for standard customer ones.

Comment (Ianmohc) [reply addresses.go:7]: Screenshot 2022-10-17 at 15 38 28

**Review by Ianmohc — CHANGES_REQUESTED** [review]

File: customer-address.go [hunk 2]
```diff
@@ -18,6 +20,41 @@ func newCustomerAddressRepository(client http.Client, createURL func(endpoint st
...
+	url := r.createURL(fmt.Sprintf("customers/%v/addresses.json", id))
+
+	addressDTO := BuildCustomerAddressDTO(address)
+
+	// Wrapping Address object inside `address` property
+	request := struct {
+		Address CustomerAddressDTO `json:"address"`
+	}{
+		Address: addressDTO,
+	}
```
Comment (rob-mohc) [inline customer-address.go:33]: What you think of this?

```suggestion
	request := struct {
		CustomerAddressDTO `json:"address"`
	}{
		addressDTO,
	}
```

File: customer-address.go [hunk 3]
```diff
@@ -18,6 +20,41 @@ func newCustomerAddressRepository(client http.Client, createURL func(endpoint st
...
+	}{}
+
+	err = json.Unmarshal(resBody, &responseDTO)
+	if err != nil {
+		return shopify.CustomerAddress{}, err
+	}
+
+	return responseDTO.CustomerAddress.ToShopify(), nil
+}
+
```
Comment (rob-mohc) [inline customer-address.go:57]: And this maybe?

```suggestion
	responseDTO := struct {
		CustomerAddressDTO `json:"customer_address"`
	}{}

	err = json.Unmarshal(resBody, &responseDTO)
	if err != nil {
		return shopify.CustomerAddress{}, err
	}

	return responseDTO.ToShopify(), nil
}

```

**Review by rob-mohc — APPROVED** [review]

**Pull request merged**